### DISCUSSION TITLE
Override Function Keys on MacOS

### DISCUSSION THREAD

**Discussion by meny** [mucommander/mucommander#1078] [Ideas]
By default the F1, F2, ... keys on MacOS devices serve various device functions, e.g. brightness, volume...

This can be permanently turned off in the system settings.

Some apps, e.g. One Commander, has a preference setting to override this system behavior and I find an option or a workaround (without having to install additional apps) would be widely welcome since the F keys play an important role in muCommander.

Thoughts, possibilities? Thank you!

**Comment by pskowronek** [comment]
> This can be permanently turned off in the system settings.

not turned off, but switched to Fn + F combination to still have that various device functions, e.g. brightness, volume.

**Comment by meny** [comment]
Yes, this is what I mentioned in part and is a very bad solution because I very often adjust brightness throughout the day and don’t want to have the function keys for the entire system at all times. My suggestion is to be able to override it just like some other Apps do. Thank you for the feedback! Sent from my iPhoneOn Nov 3, 2023, at 17:22, Piotr Skowronek ***@***.***> wrote:﻿

This can be permanently turned off in the system settings.

not turned off, but switched to Fn + F combination to still have that various device functions, e.g. brightness, volume.

**Comment by pskowronek** [comment]
Do you mean `Commander One` (https://apps.apple.com/us/app/commander-one-file-manager/id1035236694?mt=12) ?
I don't see such option over there - can you make a screenshot or sth? Or maybe it only works for mac with touch bar?

**Comment by meny** [comment]
Hi Piotr, Sorry, I wasn't at the desk.. Screenshot from OneCommander attached as requested. Also, as another example, you can scroll down the Nimble Commander feature list [1] to find a similar feature. Still.., even if others didn't have it, which they do, this is a highly useful feature option. Good day! Roman [1] https://magnumbytes.com/ On Nov 3, 2023, at 7:29 PM, Piotr Skowronek ***@***.***> wrote: Do you mean Commander One ( https://apps.apple.com/us/app/commander-one-file-manager/id1035236694?mt=12 ) ? I don't see such option over there - can you make a screenshot or sth? Or maybe it only works for mac with touch bar? — Reply to this email directly, view it on GitHub , or unsubscribe . You are receiving this because you authored the thread. Message ID: ***@***.***>

**Comment by pskowronek** [comment]
@meny I don't see screenshot attachment.

**Reply by meny** [reply]
![Screenshot 2023-11-06 at 09 31 08](https://github.com/mucommander/mucommander/assets/3902807/30f1d995-4a59-4ab1-8db7-1ccd2cb65595)

sorry mail attachments don’t seem to work

**Reply by pskowronek** [reply]
I don't have that option:
![obraz](https://github.com/mucommander/mucommander/assets/360665/4a80b91c-8fcd-4019-a337-fe19cbdacb17)


Can you tell us what macOS version you've got, whether you have touch bar on your mac.

**Reply by meny** [reply]
I'm on iMac i9 running 14.1 (23B74).
![Screenshot 2023-11-07 at 11 01 04](https://github.com/mucommander/mucommander/assets/3902807/7e6751a1-908c-4640-83dd-8f032efb85af)

**Comment by ahadas** [comment]
this is one of those features that are OS-specific (in this case for macOS) and are not available through the JRE. we have few exceptions for things like this that we implemented but generally we try to avoid such features and in this case, the benefit seems to be very limited and there are other tools that this functionality can be achieved with (I tested [Fluor](https://github.com/Pyroh/Fluor) that works great for disabling the system settings for the function keys specifically for muCommander on my laptop)
